File: Others/medium/NumberOfWaysToMakeChange.py
def numberOfWaysToMakeChange(n, denoms):
    dp = [0]*(n+1)
    dp[0] = 1

    for denom in denoms:
        for i in range(1, n + 1):
            if i - denom >= 0:
                dp[i] += dp[i - denom]

    return dp[n]


print(numberOfWaysToMakeChange(6, [1, 5]))

File: Golang/medium/number-of-ways-to-make-change.go
package main

import (
	"fmt"
)

func NumberOfWaysToMakeChange(n int, denoms []int) int {
	dp := make([]int, n+1)
	for idx := 0; idx <= n; idx++ {
		dp[idx] = 0
	}
	dp[0] = 1

	for _, denom := range denoms {
		for idx := 1; idx <= n; idx++ {
			if idx-denom >= 0 {
				dp[idx] += dp[idx-denom]
			}
		}
	}
	fmt.Printf("%v\n", dp)
	return dp[n]
}

func main() {
	n := 6
	denoms := []int{1, 5}

	NumberOfWaysToMakeChange(n, denoms)
}
